integration-tests-jvm/hashicorp-vault/src/main/java/org/apache/camel/quarkus/component/hashicorp/vault/it/HashicorpVaultResource.java

```diff
@@ -16,35 +16,75 @@
  */
 package org.apache.camel.quarkus.component.hashicorp.vault.it;
 
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
+import jakarta.ws.rs.DELETE;
 import jakarta.ws.rs.GET;
+import jakarta.ws.rs.POST;
 import jakarta.ws.rs.Path;
 import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
 import jakarta.ws.rs.core.MediaType;
 import jakarta.ws.rs.core.Response;
-import org.apache.camel.CamelContext;
-import org.jboss.logging.Logger;
+import org.apache.camel.ProducerTemplate;
 
 @Path("/hashicorp-vault")
 @ApplicationScoped
 public class HashicorpVaultResource {
+    @Inject
+    ProducerTemplate producerTemplate;
 
-    private static final Logger LOG = Logger.getLogger(HashicorpVaultResource.class);
+    @Path("/secret")
+    @POST
+    public Response createSecret(@QueryParam("key") String key, @QueryParam("value") String value) throws Exception {
+        producerTemplate.sendBody("direct:createSecret", Map.of(key, value));
+        return Response.created(new URI("https://camel.apache.org/")).build();
+    }
 
-    private static final String COMPONENT_HASHICORP_VAULT = "hashicorp-vault";
-    @Inject
-    CamelContext context;
+    @SuppressWarnings("unchecked")
+    @Path("/secret")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public Response getSecret(@QueryParam("key") String key) {
+        try {
+            Map<String, Map<String, String>> map = producerTemplate.requestBody("direct:getSecret", null, Map.class);
+            if (map.containsKey("data")) {
+                Map<String, String> data = map.get("data");
+                if (data.containsKey(key)) {
+                    return Response.ok(data.get(key)).build();
+                }
+            }
+            return Response.status(404).build();
+        } catch (Exception e) {
+            return Response.status(404).build();
```

Comment:
It's more 500 here ? Maybe depends on the context.

Reply:
> Maybe depends on the context

Yeah it's due to a quirk of the Camel component that can probably be improved.

If you try to resolve a secret that does not exist, an NPE gets thrown from the vault internals. So I figured that any exception thrown is analogous to the secret not existing, hence returning a 404.

I can rework it a bit if you prefer.

Reply:
That's fair enough for this pr @jamesnetherton :+1: